### ISSUE TITLE
Ask for width on shared-use paths

### ISSUE THREAD

**Issue by osmuser63783** [streetcomplete/StreetComplete#5445]
When out for a walk today, I was surprised to see StreetComplete does not have a quest that asks for the width of a shared-use path ([highway=cycleway/footway/path + bicycle=designated + foot=designated + segregated=no](https://wiki.openstreetmap.org/wiki/Key:segregated#Tagging)).

I would imagine that this is useful information for bicycle routing, both on shared-use sidewalks and on paths that are separate from the road network. When such paths are narrow, they are not great for cycling, especially when they are busy with pedestrians.

See also https://github.com/streetcomplete/StreetComplete/issues/3945#issuecomment-1090367765 where @westnordost suggested this.

### General
Affected tag(s) to be modified/added: [width](https://wiki.openstreetmap.org/wiki/Key:width)
Question asked: **What is the width of this shared-use path?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by any pedestrian from the outside but a survey is necessary
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

### Ideas for implementation

This can all be based on the existing quests for cycleway width and road width, so I hope the implementation effort required would be relatively small.

**Comment by osmuser63783** [comment]
On second thought, this could be used wherever `foot` and `bicycle` is `yes` or `designated` even when they do not have `segregated`, as long as they do not have `segregated=yes` (only after asking the `segregated` quest, and only for paths that don't fulfil the conditions for the cycleway width quest). It should be limited to paved paths (and only happen after the `surface` quest).

Some data
666,950x    `highway=path`/`footway`/`cycleway`, `foot=designated`, `bicycle=designated`, `segregated=no`
1,583,739x    `highway=path`/`footway`/`cycleway`, `foot=designated`/`yes`, `bicycle=designated`/`yes`, `segregated!=yes`
-1,120,462 of which have `surface`
-829,068 of which have one of the ten most common paved `surface` tags

So widening it would catch another ~160,000 or so paths.

**Comment by westnordost** [comment]
That would then be asked on just about every (hiking) path

**Comment by mnalis** [comment]
> Note that shared use paths are now excluded from the smoothness quest too

Hm, are you sure @rhhsm ?

To me it [looks like](https://github.com/streetcomplete/StreetComplete/blob/c18d8675556657288fd968d1f211a759ab9ad957/app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddPathSmoothness.kt#L12-L27) it does ask for `smoothness` on shared-use (i.e. `segregated=no`) paths, like e.g. [way 804758454](https://www.openstreetmap.org/way/804758454)

**Comment by osmuser63783** [comment]
> That would then be asked on just about every (hiking) path

Only if it's paved and legal to cycle on.

`highway=footway bicycle=yes` is a fairly common combination in the data (often without an explicit `segregated=no`). I believe that StreetComplete does ask about their `smoothness`, it would be nice to ask about their width too.

**Comment by mnalis** [comment]
> highway=footway bicycle=yes is a fairly common combination in the data

It is, but the meaning of that `bicycle=yes` is quite dubious and contested in many cases (which is why `=designated` was invented). Probably most common situation I've seen is a case of _"that is regular footway, but it has neither explicit sign disallowing bicycle, nor explicit sign allowing bicycles"_. 

And if that is common situation elsewhere too, then such tagging quickly becomes almost the equivalent of _"Add footway width"_ , which has been discussed in https://github.com/streetcomplete/StreetComplete/issues/3945.

Also note that `AddFootwayWidth.kt`  is actually [already implemented in SCEE](https://github.com/Helium314/SCEE/blob/v56.0/app/src/main/java/de/westnordost/streetcomplete/quests/width/AddFootwayWidth.kt); so if it is accepted here too, perhaps it can be reused.

**Comment by osmuser63783** [comment]
> Also note that AddFootwayWidth.kt is actually [already implemented in SCEE](https://github.com/Helium314/SCEE/blob/v56.0/app/src/main/java/de/westnordost/streetcomplete/quests/width/AddFootwayWidth.kt); so if it is accepted here too, perhaps it can be reused.

Thank you.

SCEE, even when you activate the “footway width” quest, doesn’t ask for the width of a `highway=footway bicycle=yes`, nor does it ask for the width of a `highway=path/footway/cycleway foot=designated bicycle=designated segregated=no`.

I am especially interested in tagging the width of shared but narrow paths like this:
https://www.geograph.org.uk/photo/2394166
https://www.geograph.org.uk/photo/6864411

In my experience, when there is an explicit sign saying the path is shared, the `highway=path/footway/cycleway foot=designated bicycle=designated segregated=no` is quite common. When there isn't, people come up with all sorts of tags for these paths, like `highway=footway bicycle=yes`, `highway=path bicycle=designated`, or whatever. I don't think it really matters for StreetComplete how they are tagged, as long as it's clear that bicycles are allowed (so tagging it is useful information for cycle routers), it's paved (so width is well defined and likely to be similar over long stretches of path) and not segregated.

**Comment by mnalis** [comment]
> SCEE, even when you activate the “footway width” quest, doesn’t ask for the width of a highway=footway bicycle=yes, nor does it ask for the width of a highway=path/footway/cycleway foot=designated bicycle=designated segregated=no.

@osmuser63783  Not by default, no. It is _**not** exactly_ the same as _"shared-use footway width"_ being requested here.

In SCEE, you need to enable `Expert mode` toggle in settings, and then you get  "cog" (:gear:)  icon near (almost all) quests, where you can tune it to your liking  (e.g. removing that `and bicycle !~ yes|designated` part of the filter, or whatever else you like). Don't forget to rescan for quests after changing it... 

So you can tune SCEE to your liking -- at the expense of additional complexity, of course (thus the _"SC Expert Edition"_ acronym).

Feel free to ask more questions about SCEE usage over [there](https://github.com/Helium314/SCEE/discussions) if you're interested; I mostly wanted to say here that the quite similar code is [already written](https://github.com/Helium314/SCEE/pull/462) there, so if it is decided it should be implemented in StreetComplete too, it should be relatively easy to just copy that and modify the filter slightly according to consensus here.

**Comment by rhhsm** [comment]
> > Note that shared use paths are now excluded from the smoothness quest too
> 
> Hm, are you sure @rhhsm ?
> 
> To me it [looks like](https://github.com/streetcomplete/StreetComplete/blob/c18d8675556657288fd968d1f211a759ab9ad957/app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddPathSmoothness.kt#L12-L27) it does ask for `smoothness` on shared-use (i.e. `segregated=no`) paths, like e.g. [way 804758454](https://www.openstreetmap.org/way/804758454)

Sorry for my confusion about the terminology. I indeed meant cases of a bicycle and foot path side by side. Examples https://www.openstreetmap.org/way/1182418629 and https://www.openstreetmap.org/way/6547886 (an example of the Dutch tagging tradition of seeing these as a cycle path with sidewalk)
SC asks for their surfaces, but not their smoothness.

**Comment by mnalis** [comment]
> Sorry for my confusion about the terminology. I indeed meant cases of a bicycle and foot path side by side. Examples https://www.openstreetmap.org/way/1182418629

Well, if it is `segregated=yes`, then it is complicated, as it is basically two ways "merged" into one, so:

- for `smoothness` quest that is problematic, as the parts might differ (especially in cases like that one above where one surface is `asphalt` and another is `paving_stones` - it is unlikely that they would have the same `smoothness`)
- for `width` too, it would need to be specified if the quests asks for width of whole path, or just cycleway part, or just footway part, or both (and tag appropriately).

**Comment by rhhsm** [comment]
> > Sorry for my confusion about the terminology. I indeed meant cases of a bicycle and foot path side by side. Examples https://www.openstreetmap.org/way/1182418629
> 
> Well, if it is `segregated=yes`, then it is complicated, as it is basically two ways "merged" into one

Not so complicated that it's not already asked for `surface`, which adds `cycleway:surface=*` and `footway:surface=*`. The same can be done for `smoothness` and `width`, which would add `footway:width=*`, etc. It's basically two paths side by side, and whether there's a strip of some other land in between doesn't really matter.
![Screenshot (18 jan](https://github.com/streetcomplete/StreetComplete/assets/69100427/ccaae920-f41f-40a9-8410-77421f298be0)

**Comment by westnordost** [comment]
Okay, there is a lot of off topic in this ticket.

Regarding the original request, I found this comment in the source code for `AddCyclewayWidth`

>     /* All either exclusive cycleways or ways that are cycleway + footway (or bridleway) but
>      *  segregated */

, so the exclusion of non-segregated cycle+footways was definitely deliberate.

I'd say, the reasoning is/was, that non-segregated cycle+footways are basically rather footways with lenient access restriction, category-wise, than the other way round. So, the width of these kind of ways should rather be asked in a theoretical `AddPathWidth` quest. Now, this quest does not exist in StreetComplete, but it does in SCEE, so I suggest to instead extend that quest in SCEE to make it so that  `AddCyclewayWidth`+`AddPathWidth` quests cover all foot+cyclepaths.

**Comment by rhhsm** [comment]
@westnordost the comments on my suggestion that `smoothness` should be asked for segregated cycle/foot paths are marked as "resolved". Does that mean that this will be implemented, or should I create a new issue to properly propose this?

**Comment by westnordost** [comment]
That you should create a new issue. I didn't completely follow your discussion so I wasn't sure whether to mark it as off topic or resolved.
